aips/multisig-script-payloads.md

```diff
@@ -0,0 +1,138 @@
+
+### Out of scope
+
+Not committed to changing the way multisig accounts work, or how transactions are executed. This is purely about which
+payloads can be used for a multisig account.
+
+## High-level Overview
+
+High level, there already is support for other types of payloads in both orderless and standard transactions. However,
+Script payloads were not supported from the beginning of the newer multisig account design. The goal here is to take
+advantage of existing extensibility in the framework and VM to allow script payloads to be used for a multisig.
+
+## Impact
+
+Many defi protocols have asked for script payloads around their protocols to be used in multisig accounts. This was
+specifically to allow for the flexibility of quickly changing logic at the time of submission for approval, without
+having to deploy code every time.
+
+If we don't accept this proposal, then protocols will have to continue to deploy new contracts every time they want to
+run arbitrary code through a multisig. This is a significant burden and can lead to inefficiencies in the process of
+multisig.
+
+## Alternative Solutions
+
+The alternative today is to continue to deploy new contracts every time a multisig wants to run arbitrary code.
+
+## Specification and Implementation Details
+
+1. Add a new `ScriptPayload` variant to the `MultisigPayload` enum in the Aptos core types.
+
+```rust
+ #[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+ pub enum MultisigTransactionPayload {
+     EntryFunction(EntryFunction),
+     Script(Script),
+ }
+```
+
+2. Add support in the REST API and VM to support Scripts, taking advantage of the existing functions used for it.
+3. Update the TypeScript SDK to support the new `ScriptPayload` variant, allowing users to create transactions with
+   script payloads in multisig accounts.
+4. Update indexer SDK to support the new `ScriptPayload` variant, allowing users to query transactions with script
+   payloads in multisig accounts.
+5. Update the indexer processor to handle the new type if it's doing anything special with multisig transactions.
+6. Update Petra, Aptos Connect, and other wallets to support the new `ScriptPayload` variant, allowing users to
+   create transactions with script payloads in multisig accounts.
+7. Update existing multisig wallet dapps to support the new `ScriptPayload` variant, allowing users to
+   create transactions with script payloads in multisig accounts.
+
+## Reference Implementation
+
+Aptos core code is here: https://github.com/aptos-labs/aptos-core/pull/16778
+
+## Testing
+
+> - What is the testing plan? (other than load testing, all tests should be part of the implementation details and won’t
+    need to be called out. Some examples include user stories, network health metrics, system metrics, E2E tests, unit
+    tests, etc.)
+> - When can we expect the results?
+> - What are the test results and are they what we expected? If not, explain the gap.
+
+...
```

Comment:
nit: might just be worth mentioning we'll rely on new and existing test infrastructure (e.g., unit tests, smoke tests, test networks, etc.).